dashboard/api/pkg/types/types.go

```diff
@@ -61,18 +55,61 @@ type JobModeration struct {
 	Notes         string    `json:"notes"`
 }
 
-type JobModerationSummary struct {
-	Moderation *JobModeration        `json:"moderation"`
-	Request    *JobModerationRequest `json:"request"`
-	User       *User                 `json:"user"`
+type ModerationRequest interface {
+	GetID() int64
+	GetType() ModerationType
+	GetCallback() *URL
+}
+
+type ModerationSummary struct {
+	Moderation *Moderation       `json:"moderation"`
+	Request    ModerationRequest `json:"request"`
+	User       *User             `json:"user"`
+}
+
+type JobModerationRequest struct {
+	ID       int64          `json:"id"`
+	Created  time.Time      `json:"created"`
+	Callback URL            `json:"callback"`
+	JobID    string         `json:"job_id"`
+	Type     ModerationType `json:"type"`
+}
+
+func (req *JobModerationRequest) GetID() int64 {
+	return req.ID
+}
+
+func (req *JobModerationRequest) GetType() ModerationType {
+	return req.Type
+}
+
+func (req *JobModerationRequest) GetCallback() *URL {
+	return &req.Callback
+}
+
+type JobModerationSummary = ModerationSummary
+
+// A ResultModerationRequest represents a request to moderate a result produced
+// by a job.
+//
+// We explicitly tie the moderation of results to jobs because jobs might be for
+// different users with different trust levels, and also because one job
+// producing a result that is the same as another job might be revealing
+// depending on the execution (i.e. automatically approving a result
+// representing the "empty set" might cause problems). So we tie each moderation
+// request to a moderation request for a job.
+type ResultModerationRequest struct {
+	JobModerationRequest
```

Comment:
This correctly models the relationship between the two, i.e. that a `ResultModerationRequest` *is a* `JobModerationRequest` but with extra data. Having a field would represent a `ResultModeraitonRequest` contains a or references a `JobModerationRequest` – different. Whether or not that is the best design is a conversation worth having, but certainly that is how the database and UI are trating these objects currently.

As an example of why using a field is problematic, this would marshal to a JSON object containing a moderation struct rather than embedding fields within the root object, which would require more code in the UI to process that object when they should be treated the same.